src/AgentCommerce/SalesChannel/CheckoutRoute.php

```diff
@@ -0,0 +1,94 @@
+use Swag\PayPal\AgentCommerce\Validation\CartTokenValidator;
+use Swag\PayPal\Checkout\Payment\Method\AbstractPaymentMethodHandler;
+use Swag\PayPal\Checkout\Payment\PayPalPaymentHandler;
+use Swag\PayPal\RestApi\V2\Resource\OrderResource;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Attribute\Route;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+#[Route(defaults: ['_routeScope' => ['paypal-agent'], '_agentScope' => [AgentSource::SCOPE_CART, AgentSource::SCOPE_CHECKOUT]])]
+class CheckoutRoute extends AbstractAgentCommerceRoute
+{
+    public function __construct(
+        private readonly AbstractCartOrderRoute $orderRoute,
+        private readonly CartService $cartService,
+        private readonly OrderResource $orderResource,
+        private readonly PayPalPaymentHandler $paymentHandler,
+        private readonly PayPalCartTransformer $cartTransformer,
+    ) {
+    }
+
+    #[Route('/api/paypal/v1/merchant-cart/{token}/checkout', name: 'api.paypal.merchant-cart.checkout', methods: [Request::METHOD_POST])]
+    public function checkout(string $token, Request $request, SalesChannelContext $context): AgentCartResponse
+    {
+        $extractedToken = CartTokenValidator::validateCartToken($token);
+
+        $cart = $this->cartService->getCart($extractedToken, $context);
+        if (!$cart->getLineItems()->count()) {
+            // We don't create a cart with empty items. So it must be created.
+            throw AgentException::cartNotFound($token);
+        }
+
+        $order = $this->orderRoute
+            ->order($cart, $context, new RequestDataBag($request->request->all()))
+            ->getOrder();
+
+        $primaryTransaction = $order->getTransactions()?->last();
+
+        // @deprecated tag:v11.0.0 - remove if condition with min-version of 6.7.1.0, keep content
+        if (\method_exists($order, 'getPrimaryOrderTransactionId')) {
+            $primaryTransaction = $order->getPrimaryOrderTransactionId();
+        }
+
+        if (!$primaryTransaction) {
+            throw AgentException::orderSystemError();
+        }
+
+        $body = \json_decode($request->getContent(), true, flags: \JSON_THROW_ON_ERROR);
+        $payPalOrder = $this->orderResource->get($body['payment_method']['token'], $context->getSalesChannelId());
+
+        $request->request->set(AbstractPaymentMethodHandler::PAYPAL_PAYMENT_ORDER_ID_INPUT_NAME, $payPalOrder->getId());
+
+        $payPalCart = $this->cartTransformer->convertToPayPalCart($cart, $context);
+        $payPalCart->setStatus(PayPalCart::STATUS__COMPLETE);
+        $payPalCart->setPaymentMethod($this->createPaymentMethod($payPalOrder->getId()));
+
+        try {
+            // @phpstan-ignore new.deprecated
+            $payment = new AsyncPaymentTransactionStruct($primaryTransaction, $order, '');
+            $response = $this->paymentHandler->pay($payment, new RequestDataBag($request->request->all()), $context);
```

Comment:
In 6.6.x, the payment handler either redirects back to shopware and the finalize action, or to PayPal. We need to differentiate there. If the redirect goes to PayPal, we need to put the Approval URL, if we redirect to finalize, we need to call finalize to capture.